### ISSUE TITLE
Hiding event content

### ISSUE THREAD

**Issue by jladbury** [almende/vis#532]
I want to provide a tabbed interface giving a different view of a timeline on each of two tabs, and to give the user a choice of showing or hiding the events' content on the active tab (when hidden, just the markers are shown). I have basically got this working, but am having trouble redrawing the timeline, particularly when going from a 'hidden' state to a 'visible one'.

I have made 3 jsBins to investigate and illustrate the problem. Each of them includes some explanation and instructions.

 http://jsbin.com/qujudo/2/ shows the problems: a 'small' one and a 'big' one, described in its comments 
 http://jsbin.com/qujudo/3/ shows an alleviation for the problem, which introduces its own insofar as the jQuery fade effects are not shown
 http://jsbin.com/qujudo/1/ shows the 'small' problem mentioned above in a page which does not use tabs. 

The tabs are provided by jQWidgets (ref: http://www.jqwidgets.com/jquery-widgets-documentation/documentation/jqxtabs/jquery-tabs-getting-started.htm). 

All comments appreciated! I have not yet raised this with jQWidgets, as your comments about the third bin (confusingly ' http://jsbin.com/qujudo/1/' - sorry about that!) might help resolve the others too.

**Comment by josdejong** [comment]
The problem here originates from adjusting the internal DOM of the Timeline using jQuery, without the Timeline knowing that something has changed. For example, the Timeline calculates the size of an item only once and caches that.

To solve this, you should do updates in the contents and styling of the items by updating their data via a `DataSet`: updating and items `content` and `className` fields. 

To force the timeline to recalculate an items size, you could create an update sequence field in the items data and increment that. When the data of an item changes, the Timeline marks it as dirty and will recalculate and update its size and contents.

**Comment by jladbury** [comment]
Thanks for looking into this. I have a couple of observations:

1. I tried adding and updating a new field (not part of the HTML content) and also tried updating the content, but neither seemed to make any difference. I wonder if I have misunderstood you - if you get time, bins http://jsbin.com/qujudo/14/ (added seq field) and http://jsbin.com/qujudo/13/ (updated content).

2. I would prefer not to update the dataset in order to update the display - may I remind you of the first sentence of this issue in case it gets forgotten: _'I want to provide a tabbed interface giving a different view of a timeline on each of two tabs'_.

To achieve this, my application adds a number of classes to each item as it populates a vis dataset from a JSON file, and then allows the user to alter the display of the items on each tab by manipulating the underlying CSS for those classes. (Notwithstanding my use of jQuery in the bins, I mostly use the **jss** library https://github.com/Box9/jss library to do this. Unlike jQuery, which adds CSS to the DOM, this library updates the CSS rules themselves). 

I have done it this way for two reasons. Firstly I assume that for a large dataset it will be more efficient to update the CSS than to recalculate the HTML content of an item (maybe I am wrong about this). The second reason for doing it this way is that when the user updates an item, I only have to reflect that in a single dataset. It would be unwieldy to have to update separate datasets for each view. And in future I intend to allow the user to add extra tabs and views dynamically . . . 

Once again your comments would be appreciated. I am particularly anxious that while I have got most of my architecture working OK - it works perfectly for such things as highlighting and otherwise formatting individual events and parts of events, failing as described only when the event size changes - if the approach is out of kilter with your own future direction I might be at risk of it suddenly failing when you produce a new version.
